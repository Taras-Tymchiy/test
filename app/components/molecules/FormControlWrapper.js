// @flow
import * as React from 'react';
import { View, StyleSheet, ActivityIndicator, Text, Button, Modal, TouchableOpacity } from 'react-native';

import Header from './Header';
import { TextRegular, TextSecondary }  from '../atoms/Text';
import ScreenView from '../atoms/ScreenView';
import Icon from '../atoms/Icon';

import { COLOR_ELEMENT_BG, SPACING_S, SPACING_M, COLOR_ELEMENT_BORDER } from '../styles/constants';
import arrow from '../../assets/ios7-arrow-right.png';
import styles from './FormControlWrapper.styles'; 

export type FormControlWrapperProps = {
  title: string;
  description: string;
  value: any;
  children: React.Node;
  onSave: () => void;
  onDiscard: () => void;
}

export type FormControlWrapperState = {
  modalVisible: boolean;
}

export default class FormControlWrapper 
  extends React.PureComponent<FormControlWrapperProps, FormControlWrapperState> {
    state = {
      modalVisible: false
    };

    openModal = ()=> {
      this.setState({modalVisible:true});
    }

    closeModal = ()=> {
      const { onDiscard } = this.props;
      if (onDiscard) {
        onDiscard();
      }
      this.setState({modalVisible:false});
    }

    saveChanges = ()=> {
      const { onSave } = this.props;
      if (onSave) {
        onSave();
      }
      this.setState({modalVisible:false});
    }

    render() {
      const { title, value, description } = this.props;
      
      return (
        <View>
          <TouchableOpacity 
            style={styles.mainContainer}
            onPress={this.openModal}>
            <View style={styles.textRow}>
              <TextRegular>{title}</TextRegular>
              <TextSecondary>{value}</TextSecondary>
            </View>
            <Icon source={arrow} />
          </TouchableOpacity>
          <Modal
              visible={this.state.modalVisible}
              animationType={'slide'}
              onRequestClose={this.closeModal} 
            >
              <Header title={title}
                leftButtonTitle="Back" rightButtonTitle="Done" 
                onLeftButtonPress={this.closeModal} 
                onRightButtonPress={this.saveChanges} 
              />
              <ScreenView>
                <View style={styles.descriptionContainer}>
                  <Text>{description}</Text>
                </View>
                <View style={styles.inputContainer}>
                  {this.props.children}
                </View>
              </ScreenView>
            </Modal>
        </View>
      );
    }
}
