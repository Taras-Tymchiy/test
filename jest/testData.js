import { Post } from "../app/entities/Post";


export const posts: Post[] = [
    {
        id: 1,
        id_str: '1',
        created_at: '2015-01-01',
        user: {
            id: 1,
            name: 'John Snow',
        },
        text: 'post text 1'
    },
    {
        id: 2,
        id_str: '2',
        created_at: '2015-01-02',
        user: {
            id: 2,
            name: 'Barack Obama',
        },
        text: 'post text 2'
    }
]