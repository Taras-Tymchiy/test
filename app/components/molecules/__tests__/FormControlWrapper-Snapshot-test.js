import React from 'react';
import * as jest from 'jest';
import renderer from 'react-test-renderer';
import FormControlWrapper from '../FormControlWrapper';
import { View } from 'react-native';

jest.mock('../../../assets/ios7-arrow-right.png', ()=>({ uri: 'ios7-arrow-right.png'}));
jest.mock('../Header', (...props)=> `Header: ${props}`);

const props: FormControlWrapperProps = {
    value: 'testValue',
    title: 'Test title',
    description: 'test description'
};

it('renders a FormControlWrapper using Snapshots', () => {
  expect(renderer.create(
    <FormControlWrapper {...props} />
  )).toMatchSnapshot();
});
