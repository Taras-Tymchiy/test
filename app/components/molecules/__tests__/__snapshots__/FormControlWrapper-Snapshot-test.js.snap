// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a FormControlWrapper using Snapshots 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderBottomWidth": 0.5,
        "borderColor": "#CCC",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 12,
        "paddingRight": 6,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginRight": 6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 16,
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        Test title
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#A0A0A0",
              "fontSize": 14,
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        testValue
      </Text>
    </View>
    <Image
      resizeMethod="scale"
      resizeMode="center"
      source={
        Object {
          "uri": "ios7-arrow-right.png",
        }
      }
      style={
        Object {
          "height": 16,
          "opacity": 0.5,
          "width": 16,
        }
      }
    />
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    visible={false}
  >
    <Header: 
      leftButtonTitle="Back"
      onLeftButtonPress={[Function]}
      onRightButtonPress={[Function]}
      rightButtonTitle="Done"
      title="Test title"
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#EEE",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          test description
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderBottomWidth": 0.5,
            "borderColor": "#CCC",
            "borderTopWidth": 0.5,
            "padding": 12,
          }
        }
      />
    </View>
  </Modal>
</View>
`;
